src/ipc.rs

```diff
@@ -342,16 +342,16 @@ impl OpaqueIpcMessage {
         }
     }
 
-    pub fn to<T>(mut self) -> Result<T, bincode::Error> where T: Deserialize + Serialize {
+    pub fn to<T>(mut self) -> Result<T, bincode::Error> where T: for<'de> Deserialize<'de> + Serialize {
         OS_IPC_CHANNELS_FOR_DESERIALIZATION.with(|os_ipc_channels_for_deserialization| {
             OS_IPC_SHARED_MEMORY_REGIONS_FOR_DESERIALIZATION.with(
                     |os_ipc_shared_memory_regions_for_deserialization| {
                 mem::swap(&mut *os_ipc_channels_for_deserialization.borrow_mut(),
                           &mut self.os_ipc_channels);
                 mem::swap(&mut *os_ipc_shared_memory_regions_for_deserialization.borrow_mut(),
                           &mut self.os_ipc_shared_memory_regions);
-                let mut data = &*self.data;
-                let mut deserializer = bincode::Deserializer::new(&mut data, bincode::Infinite);
+                let data = bincode::read_types::SliceReader::new(&self.data[..]);
+                let mut deserializer = bincode::Deserializer::new(data, bincode::Infinite);
```

Comment:
Shouldn't the binding better be called something else than `data` after this change?... (`reader` perhaps? Or maybe `data_reader`?...)